tests/31sync/06state.pl

```diff
@@ -1,4 +1,30 @@
-test "State is included in the initial sync",
+use Future::Utils qw( repeat );
+
+# call /sync repeatedly until it returns a result
+# with an event in the given room
+sub wait_for_event_in_room {
```

Comment:
For expediency in getting this committed now, I'd say leave it how it is. But perhaps put a TODO note in to suggest finding a way to run it using `await_event_for()` instead, so that can be neatened up at some future time.

Mmm.. technical debt.. ;)